lib/reservations.rb

```diff
@@ -0,0 +1,123 @@
+require 'date'
+
+module Hotel
+  class RoomQuantityError < StandardError
+  end
+
+  class Reservations
+    attr_reader :all_rooms, :all_reservations
+
+    def initialize
+      @all_rooms = []
+      create_rooms
+
+      @all_reservations = []
+
+
+    end
+
+    def create_rooms
+      20.times do |i|
+        num = i + 1
+        @all_rooms << Room.new(num)
+      end
+    end
+
+
+    def make_booking(no_of_rooms, start_date, end_date, block: false)
+
+      date_range = DateRange.new(start_date, end_date)
```

Comment:
I like the idea of both block and individual reservations going through this method - good work keeping things DRY.